### PR TITLE
MEN-2248: Set production/demo intervals conditionally

### PR DESCRIPTION
**Pull request by lluiscampos**
Select demo default intervals only when user specifies --demo-host-ip,
else use production defaults (updated in parallel in MEN-2248)

Changelog: Title

Signed-off-by: Lluis Campos <lluis.campos@northern.tech>

### CODE REVIEW

**Comment by lluiscampos** [comment]
I submitted the pull request before getting all the information from the P.O. The JIRA issue has been clarified and the changes in this PR need some rework. See updated description in MEN-2248
Basically, we want to use a new flag --demo instead of using the --demo-host-ip
Sorry for the inconvenience.

**Comment by apodogrocki** [comment]
@lluiscampos Ok, so waiting for final clarification from @estenberg .

**Comment by lluiscampos** [comment]
Hi @apodogrocki 
I have reworked the code following the latest description in JIRA. Please have a look again.
As I have implemented it, --demo will require --demo-host-ip but not vice-versa. I couldn't decide on how to handle all the related demo params that we can potentially add.

File: convert-stage-4.sh [hunk 1]
```diff
@@ -8,13 +8,14 @@ Mender executables, service and configuration files installer.
 Usage: $0 [options]
 
     Options: [-m|--mender-disk-image | -g|--mender-client | -a|--artifact-name |
-              -d|--device-type | -p|--demo-host-ip | -u| --server-url |
+              -d|--device-type | -e|--demo | -p|--demo-host-ip | -u| --server-url |
```
Comment (apodogrocki) [inline convert-stage-4.sh]: '-e' option is reserved in the main 'mender-convert' script for 'storage-total-size-mb' - so maybe it is worth to not mix that

File: mender-convert [hunk 2]
```diff
@@ -33,7 +33,7 @@ Expert commands:
                                                   related files
 
 Options: [-r|--raw-disk-image | -m|--mender-disk-image | -s|--data-part-size-mb |
-          -d|--device-type | -p|--rootfs-partition-id | -i|--demo-host-ip |
+          -d|--device-type | -p|--rootfs-partition-id | -e|--demo | -i|--demo-host-ip |
```
Comment (apodogrocki) [inline mender-convert]: Conflict between '--demo' and '--storage-total-size-mb' - both use short name as '-e'

Comment (lluiscampos) [reply mender-convert]: Good catch, thank you.

**Pull request merged**